File: fixtures/dogtinder-android/app/src/main/java/com/natively/dogtinder/Dog.kt
package com.natively.dogtinder

import androidx.compose.ui.graphics.Color

data class Dog(
    val id: String,
    val name: String,
    val age: Int,
    val breed: String,
    val distanceKm: Int,
    val bio: String,
    val mutualMatch: Boolean,
    val color: Color,
)

val dogs: List<Dog> = listOf(
    Dog("biscuit", "Biscuit", 3, "Corgi", 2, "Loves belly rubs and long naps in the sun.", true, Color(0xFFF4A261)),
    Dog("rex", "Rex", 5, "German Shepherd", 4, "Serious about fetch. Not serious about anything else.", false, Color(0xFF264653)),
    Dog("mochi", "Mochi", 1, "Shiba Inu", 1, "Small, fluffy, judging you.", false, Color(0xFFE9C46A)),
    Dog("luna", "Luna", 4, "Husky", 7, "Will sing you the song of her people.", true, Color(0xFF2A9D8F)),
    Dog("bruno", "Bruno", 6, "Boxer", 3, "Gentle giant with a drool problem.", false, Color(0xFF8D5B4C)),
    Dog("pepper", "Pepper", 2, "Dalmatian", 5, "Spots on the outside, chaos on the inside.", false, Color(0xFF6D6875)),
)

fun dogById(id: String): Dog = dogs.first { it.id == id }

File: fixtures/dogtinder-ios/DogTinder/Dog.swift
import SwiftUI

struct Dog: Identifiable, Hashable {
    let id: String
    let name: String
    let age: Int
    let breed: String
    let distanceKm: Int
    let bio: String
    let mutualMatch: Bool
    let colorHex: UInt32

    var color: Color {
        Color(
            red: Double((colorHex >> 16) & 0xFF) / 255.0,
            green: Double((colorHex >> 8) & 0xFF) / 255.0,
            blue: Double(colorHex & 0xFF) / 255.0
        )
    }

    var nameAndAge: String { "\(name), \(age)" }
    var distanceText: String { "\(distanceKm) km away" }

    static let all: [Dog] = [
        Dog(id: "biscuit", name: "Biscuit", age: 3, breed: "Corgi", distanceKm: 2,
            bio: "Loves belly rubs and long naps in the sun.", mutualMatch: true, colorHex: 0xF4A261),
        Dog(id: "rex", name: "Rex", age: 5, breed: "German Shepherd", distanceKm: 4,
            bio: "Serious about fetch. Not serious about anything else.", mutualMatch: false, colorHex: 0x264653),
        Dog(id: "mochi", name: "Mochi", age: 1, breed: "Shiba Inu", distanceKm: 1,
            bio: "Small, fluffy, judging you.", mutualMatch: false, colorHex: 0xE9C46A),
        Dog(id: "luna", name: "Luna", age: 4, breed: "Husky", distanceKm: 7,
            bio: "Will sing you the song of her people.", mutualMatch: true, colorHex: 0x2A9D8F),
        Dog(id: "bruno", name: "Bruno", age: 6, breed: "Boxer", distanceKm: 3,
            bio: "Gentle giant with a drool problem.", mutualMatch: false, colorHex: 0x8D5B4C),
        Dog(id: "pepper", name: "Pepper", age: 2, breed: "Dalmatian", distanceKm: 5,
            bio: "Spots on the outside, chaos on the inside.", mutualMatch: false, colorHex: 0x6D6875),
    ]
}
